aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -481,6 +481,20 @@ public void query(
                 final SQLiteTable sqliteTable = SQLiteTable.fromSchema(modelSchema);
                 final String primaryKeyName = sqliteTable.getPrimaryKeyColumnName();
 
+                if (!dataExistsInSQLiteTable(sqliteTable.getName(), primaryKeyName, item.getId())) {
```

Comment:
Two questions here:

Does this mean we perform two operations on SQL? A lookup and the a delete? Performance wise that’s probably okay for our delete API. Might have implications for atomicity of the operation.

Second more important - should we fire an event on the observe method here? Would the sync engine care about observing this case?

Reply:
> Does this mean we perform two operations on SQL? A lookup and the a delete?

This lookup is already performed in `save` to determine whether item should be created or updated. There hasn't been an issue there so far, so it will be fine.

> should we fire an event on the observe method here? 

No change took place in the database, so I don't see the need for the sync engine to worry about it.